include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -571,13 +586,31 @@ class permutation_iterator
         return my_source_it;
     }
 
+    template <typename _T = IndexMap, typename ::std::enable_if<!__is_functor<_T>, int>::type = 0>
```

Comment:
It is a SFINAE approach. `int` just of dummy type parameter here. For example, please see `nonorange`  library (\oneDPL\include\oneapi\dpl\pstl\ranges\nanorange.hpp).
The second template parameter is well formed (if T is a functor) or ill formed, In case of ill formed  a compiler has a substitution  failure , which is not a error (SFINAE)  and "drop off"  this method form a class.